tools/cli/admin_task_list_commands.go

```diff
@@ -137,35 +142,74 @@ func AdminListTaskList(c *cli.Context) error {
 		return commoncli.Problem("Operation GetTaskListByDomain failed.", err)
 	}
 
+	if c.String(FlagFormat) == formatJSON {
+		prettyPrintJSONObject(getDeps(c).Output(), response)
+		return nil
+	}
+
 	fmt.Println("Task Lists for domain " + domain + ":")
-	table := []TaskListRow{}
+	tlByName := make(map[string]TaskListRow)
 	for name, taskList := range response.GetDecisionTaskListMap() {
-		table = append(table, TaskListRow{name, "Decision", len(taskList.GetPollers())})
+		row := tlByName[name]
+		row.Name = name
+		row.DecisionPollerCount = len(taskList.GetPollers())
+		tlByName[name] = row
 	}
 	for name, taskList := range response.GetActivityTaskListMap() {
-		table = append(table, TaskListRow{name, "Activity", len(taskList.GetPollers())})
-	}
+		row := tlByName[name]
+		row.Name = name
+		row.ActivityPollerCount = len(taskList.GetPollers())
+		tlByName[name] = row
+	}
+	table := maps.Values(tlByName)
+	slices.SortFunc(table, func(a, b TaskListRow) int {
+		return strings.Compare(a.Name, b.Name)
+	})
```

Comment:
I _think_ you'll want `slices.SortStableFunc` here, or sort by both name and type.  `Sort` is unstable, so you might see activity/decision reorders under the same name.

e.g. I think this is possible:
```
decision  tasklist-A   3
activity  tasklist-A   3
activity  tasklist-B   3
decision  tasklist-B   3
```

I don't think the tests exercise this, so it's rather minor at the moment, but probably still worth doing.

Reply:
I actually changed the table format entirely, it's now this:

| Name | Activity Pollers | Decision Pollers |
| :--:| :---: | :---: | 
| tasklist-A | 3 | 3 |
| tasklist-B | 3 | 3 |

So the name is unique and we've aggregated the counts across both types. My goal was to align it more with the Cadence UI, which lists the pollers of a TaskList and has a checkbox for whether each has a Decision/Activity poller.

Reply:
ah, I see, it's just iteratively filling the same row in both loops.

yea that's fine then 👍